src/main/scala/projectEuler/LargestPrimeFactor.scala

```diff
@@ -0,0 +1,26 @@
+package projectEuler
+
+import scala.math._
+
+/**
+ * Problem 3. Largest prime factor
+ * The prime factors of 13195 are 5, 7, 13 and 29.
+ * What is the largest prime factor of the number 600851475143 ?
+ * @see https://projecteuler.net/problem=3
+ */
+class LargestPrimeFactor(num:Long) {
+
+  private def isFactor(x: Long): Boolean = {
+    num % x == 0
+  }
+
+  private def isPrime(x: Long): Boolean = {
+    2L until x forall( y => x % y != 0)
+  }
+
+  lazy val primeFactors = (2L to sqrt(num).toLong)
```

Comment:
If it is a private method you should not and cannot add a unit test on it - it should be tested via the public methods that use it. This is because it is a private helper function which should be able to change without breaking the external API.